src/training/train.cpp

```diff
@@ -157,27 +171,85 @@ void TestInference(RecordInput* input, GraphInference* inference) {
 
 }
 
-void Train(RecordInput* input, GraphInference* inference) {
-  inference->SSVMInit(FLAGS_regularization_const, FLAGS_svm_margin);
+void Train(RecordInput* input, GraphInference* inference, int fold_id) {
```

Comment:
This method is full of if statements about PL / SSVM. There is really not much benefit in having one method and not two for the two cases. It also makes it quite bad with the combination of PL+SSVM (if it would be useful in the end).